### PR TITLE
Publish `mondrian-charts` and `@fiberplane/fiberplane-charts`

### PR DESCRIPTION
**Pull request by arendjr**
# Description

This sets up `mondrian-charts` for publication to Crates.io and `@fiberplane/fiberplane-charts` and `@fiberplane/prometheus-query` for publication to NPM.

The main changes are a lot of documentation added and the removal of the `dist` folder from `fiberplane-charts`. We will have to see if the latter really makes for a more pleasant dev experience (I know nobody likes regenerating the `dist` folder, but requiring releases before we can use new features downstream *might* be even worse?).

Also contains some smaller tweaks and a bit of API streamlining that I ran into as I was documenting things. This also introduces a few backwards incompatibilities again, but fortunately nobody depends yet on the Rust crate, and for the TypeScript package it was inevitable due to the name change (added the `@fiberplane` namespace).

# Checklist

- :warning: **The changes are not backwards compatible.**
- ~~[ ] The OpenAPI schema and generated client have been updated.~~
- ~~[ ] New models module has been added to api generator xtask~~
- [x] New types/fields are [well-documented and annotated](/CONTRIBUTING.md#adding-types-and-their-annotations).
- ~~[ ] The CHANGELOG is updated.~~

> Please link any related PRs in other repositories that depend on this:

After merging, please merge related PRs ASAP, so others don't get blocked.

### CODE REVIEW

**Comment by brettimus** [comment]
regarding the local dev experience for charts:

it'd be good to document how we expect people to develop locally, because we're still going to be doing that quite a bit. 

i think it'd still be possible to generate a build of the charts and link to it locally, which is how i've been doing things anyhow most of the time. that said, it is a lot of _steps_:

- run `yarn build` in fiberplane-charts
- run `yarn pack --out charts-zzz.tgz`
- modify package.json in e.g., dora to point fiberplane-charts to the `charts-zzz.tgz` file

maybe there's a way to make this more seamless? like a script that wraps up all three steps?

File: fiberplane-charts/.npmignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+node_modules/
```
Comment (brettimus) [inline fiberplane-charts/.npmignore]: when `.npmignore` is present, `.gitignore` is not used at all. for safety, we might wanna add `.vscode` to this as well, or any other local files that might find their way into the published package (granted, if we build + publish from ci, this is less of a concern)

File: fiberplane-charts/README.md [hunk 2]
```diff
@@ -4,22 +4,38 @@
...
+    "@fiberplane/fiberplane-charts": "https://git@github.com/fiberplane/fiberplane.git#workspace=fiberplane-charts&head=<branch_name>"
+  }
 ```
+
+Run `yarn` to install as usual.
+
+If you want to update `fiberplane-charts` while using the above syntax, you may
+use `yarn up "@fiberplane/fiberplane-charts@<full-dependency-url>"`.
+
+Make sure to remove the `dist/` folder before opening a PR.
```
Comment (brettimus) [inline fiberplane-charts/README.md]: this relates to [my comment about the local dev experience](https://github.com/fiberplane/fiberplane/pull/96#issuecomment-1683578058)

this upside of what you've written here is it makes the temporary build of charts easier to share across multiple people working on the same PR.

however, manually removing `dist` does not seem ideal.

the local dev workflow i outlined has the benefit that `dist` stays out of git; however, if you wanted someone to check your work, then they would have to repeat the process of creating a local build and linking to it

curious about your thoughts

File: fiberplane-charts/.npmignore (hunk 1 above)
Comment (arendjr) [reply fiberplane-charts/.npmignore]: Good call!

File: fiberplane-charts/README.md (hunk 2 above)
Comment (arendjr) [reply fiberplane-charts/README.md]: Manually removing `dist` is not ideal indeed, but at least I think it's safe enough now that CI will fail if you check it in. So you *can* check it in if you want to share it with someone to let them try it out, but CI will complain so you wouldn't merge it into `main` at least.

But I agree your `yarn pack` workflow should be the one we recommend by default.

**Review by brettimus — APPROVED** [review]

**Review by mellowagain — APPROVED** [review]

**Pull request merged**